crates/query-engine/translation/src/translation/helpers.rs

```diff
@@ -315,26 +315,17 @@ impl State {
         sql::ast::TableReference::AliasedTable(alias)
     }
 
-    /// Fetch the tracked native queries used in the query plan and their table alias.
-    pub fn get_native_queries(self) -> Vec<NativeQueryInfo> {
-        self.native_queries.native_queries
-    }
-
-    /// increment the table index and return the current one.
-    fn next_global_table_index(&mut self) -> TableAliasIndex {
-        let TableAliasIndex(index) = self.global_table_index;
-        self.global_table_index = TableAliasIndex(index + 1);
-        TableAliasIndex(index)
+    /// Fetch the tracked native queries used in the query plan and their table alias,
+    /// and the global table index.
+    pub fn get_native_queries_and_global_index(self) -> (Vec<NativeQueryInfo>, TableAliasIndex) {
```

Comment:
We break down the state so we can consume the native queries when processing native queries, and use the global table alias index outside of it when inventing names for the wrapping ctes.